File: packages/ruviz-web/examples/boxplot.ts
import { createPlot } from "../src/index.js";
import { sampleDistribution } from "./data.js";
import type { PlotExampleDefinition } from "./types.js";

export const example: PlotExampleDefinition = {
  slug: "boxplot",
  title: "Boxplot",
  summary: "Quartiles and outliers summarized as a boxplot.",
  category: "statistical",
  createPlot() {
    return createPlot().sizePx(640, 320).title("Boxplot").ylabel("value").boxplot(sampleDistribution());
  },
};

File: python/examples/boxplot.py
from __future__ import annotations

from _shared import ExampleMeta, base_plot, sample_distribution, save_example

META = ExampleMeta(
    slug="boxplot",
    title="Boxplot",
    summary="Quartiles and outliers summarized as a boxplot.",
    section="Statistical plots",
)


def build_plot():
    return base_plot("Boxplot").ylabel("value").boxplot(sample_distribution())


if __name__ == "__main__":
    save_example(META, build_plot())
